src/sgame/sg_bot_ai.cpp

```diff
@@ -1023,11 +1017,11 @@ AINodeStatus_t BotActionRush( gentity_t *self, AIGenericNode_t *node )
 		return STATUS_FAILURE;
 	}
 
-	if ( !GoalInRange( self, 100 ) )
```

Comment:
I think BotGetGoalRadius is about the size of the goal entity, while GoalInRange takes any distance away from the goal you want to be. So using BotGetGoalRadius is the desired range basically means you want to be touching the goal entity. This would be bad if you are a human and your goal is an acid tube - better to shoot from a safe distance than to go touch it.

Reply:
Except that "action rush" is not about fighting at all. When a bot sees a target, "action fight" is triggered instead (because alertedToEnemy).
Without this change, it means that if action rush takes a building which is bigger than 100 (currently does not exist, ofc)  with "don't touch building" cvar bots will be stuck trying to move to center.
This change makes things more robust, generally speaking.

Reply:
> When a bot sees a target, "action fight" is triggered instead (because alertedToEnemy).

Well, in principle it's possible to write the BT differently.

How about like `if ( !GoalInRange( self, 30 + BotGetGoalRadius(target) ) )` ?

Reply:
As for the problem of acid tube, or spiker, or turrets even... action rush leads bot to either the unique or a spawn. Never to a defensive structure. When the bot becomes alerted to enemy, then action fight defines the new goal, which is in practice a major issue (prevents fleeing while fighting, notably), that I want to fix later.

What *can* be done, and I have done it in my mod, is to use a `concurrent` node when fighting, so that when fighting, it target is  a building, then move backward as long as target is in range.
Of course, this "solution" create new problems, because the range information of weapons is completely stupid. Also, this only works because of my memory rewrite, as otherwise fighting something hidden by a door would make said thing no longer a target, as the target is removed from the stack.
Still, this is a better solution that the current situation where bots just kiss acid tubes if they happen to have be close enough when they decided to target them.

Reply:
OK, in any case it's an improvement over the status quo which just returns STATUS_RUNNING forever.

Since BotMoveToGoal already checks `GoalInRange( self, BotGetGoalRadius( self ) )`, you can delete this if statement - it's duplicated code.

Reply:
Said Ok on IRC.